src/web/mod.rs

```diff
@@ -238,29 +260,39 @@ fn match_version(conn: &Connection, name: &str, version: Option<&str>) -> MatchV
         })
         .unwrap_or_else(|| "*".into());
 
+    let mut corrected_name = None;
     let versions: Vec<(String, i32)> = {
-        let query = "SELECT version, releases.id
+        let query = "SELECT name, version, releases.id
             FROM releases INNER JOIN crates ON releases.crate_id = crates.id
-            WHERE name = $1 AND yanked = false";
+            WHERE normalize_crate_name(name) = normalize_crate_name($1) AND yanked = false";
         let rows = conn.query(query, &[&name]).unwrap();
-        if rows.is_empty() {
-            return MatchVersion::None;
-        }
-        rows.iter().map(|row| (row.get(0), row.get(1))).collect()
+        let mut rows = rows.iter().peekable();
+
+        if let Some(row) = rows.peek() {
+            let db_name = row.get(0);
+            if db_name != name {
+                corrected_name = Some(db_name);
+            }
+        };
```

Comment:
This removes the `is_empty()` check, was that intentional? I think the behavior should be the same, but it would be best to add a test if you want to keep this change.

Reply:
Yes, there is very minimal extra work that would occur without the early exit, so it seems cleaner to just let it fall-through. I'll add a test.